modules/rgbd/src/tsdf.cpp

```diff
@@ -394,21 +422,20 @@ struct IntegrateInvoker : ParallelLoopBody
                     // norm(camPixVec) produces double which is too slow
                     float pixNorm = sqrt(camPixVec.dot(camPixVec));
                     // difference between distances of point and of surface to camera
-                    TsdfType sdf = pixNorm*(v*dfac - camSpacePt.z);
-
+                    float sdf = pixNorm*(v*dfac - camSpacePt.z);
                     // possible alternative is:
                     // kftype sdf = norm(camSpacePt)*(v*dfac/camSpacePt.z - 1);
                     if(sdf >= -volume.truncDist)
                     {
-                        TsdfType tsdf = fmin(1.f, sdf * truncDistInv);
+                        TsdfType tsdf = floatToTsdf(fmin(1.f, sdf * truncDistInv));
 
                         TsdfVoxel& voxel = volDataY[z*volume.volDims[2]];
                         WeightType& weight      = voxel.weight;
                         TsdfType& value  = voxel.tsdf;
 
                         // update TSDF
-                        value  = (value*weight+tsdf) / (weight + 1);
-                        weight = min(weight + 1, volume.maxWeight);
+                        value  = floatToTsdf((tsdfToFloat(value)*weight+ tsdfToFloat(tsdf)) / (weight + 1));
+                        weight = min(int(weight + 1), int(volume.maxWeight));
```

Comment:
Need to make sure that `tsdf_max_weight` < 255 always, and document it.

Reply:
See line 40, there was a special assert added for that. And yes, this should be documented.